src/file-upload/file-item.tsx

```diff
@@ -0,0 +1,170 @@
+    onEdit,
+}: Props) => {
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const isLoading = progress < 1;
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleDelete = () => {
+        onDelete();
+    };
+
+    const handleEdit = () => {
+        if (onEdit) onEdit();
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderActionButton = () => {
+        if (isLoading) {
+            return (
+                <LoadingActionContainer>
+                    <ProgressBar progress={progress} />
+                </LoadingActionContainer>
+            );
+        } else if (errorMessage) {
+            return (
+                <ItemActionContainer>
+                    <ErrorIconButton
+                        onClick={handleDelete}
+                        data-testid={`${id}-error-delete-button`}
+                    >
+                        <CrossIcon />
+                    </ErrorIconButton>
+                </ItemActionContainer>
+            );
+        } else {
+            const isEditable = isSupportedImageType(file.type);
+
+            return (
+                <ItemActionContainer $hasEditButton={isEditable}>
+                    {/* TODO: Add in part 2 */}
+                    {/* {isEditable && (
+                        <IconButton
+                            key="edit"
+                            data-testid={`${id}-edit-button`}
+                            type="button"
+                            styleType="light"
+                            onClick={handleEdit}
+                        >
+                            <PencilIcon />
+                        </IconButton>
+                    )} */}
+                    <IconButton
+                        key="delete"
+                        data-testid={`${id}-delete-button`}
+                        type="button"
+                        styleType="light"
+                        onClick={handleDelete}
+                    >
+                        <BinIcon />
+                    </IconButton>
+                </ItemActionContainer>
+            );
+        }
+    };
+
+    return (
+        <Item
+            id={id}
+            $error={!!errorMessage}
+            $loading={isLoading}
+            $editable={isSupportedImageType(file.type)}
+        >
+            <Content>
+                <ItemNameSection>
+                    <ItemText
+                        data-testid="name"
+                        weight={description ? "semibold" : "regular"}
+                    >
+                        {file.name}
```

Comment:
will need to truncate the text if the filename exceeds two lines

Reply:
Was intending for truncations to be done in the later portions. But i guess i could add them in too